lms/services/canvas_api/_pages.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass
```

Comment:
Adding a new file/module for just pages.

This includes the dataclass, schema and actual implementation.  The main client module it's getting confusing with lots of inlined classes.